File: apps/Offline-AI-Buddy/android/core/src/main/java/com/americangroupllc/offlineaibuddy/core/models/ChatSession.kt
package com.americangroupllc.offlineaibuddy.core.models

import kotlinx.serialization.Serializable

@Serializable
data class ChatMessage(
    val id: String,
    val role: Role,
    val text: String,
    val tsMillis: Long = System.currentTimeMillis(),
) {
    @Serializable
    enum class Role { USER, ASSISTANT, SYSTEM }
}

@Serializable
data class ChatSession(
    val id: String,
    val profileId: String,
    val language: Language,
    val kind: Kind,
    val messages: List<ChatMessage> = emptyList(),
    val startedAtMillis: Long = System.currentTimeMillis(),
) {
    @Serializable
    enum class Kind(val displayName: String, val availableInKidSafe: Boolean) {
        CHAT("Chat", true),
        ROAST("Roast", false),
        DAILY_CHALLENGE("Daily Challenge", true),
        PARTY_QUESTIONS("Party Questions", true),
        GAME_COACH("Game Coach", true),
        TRANSLATE("Translate", true),
    }
}

File: apps/Offline-AI-Buddy/shared/BuddyAICore/Sources/BuddyAICore/Models/ChatSession.swift
import Foundation

public struct ChatSession: Codable, Hashable, Sendable, Identifiable {
    public let id: UUID
    public let profileId: UUID
    public var language: Language
    public var kind: Kind
    public var messages: [ChatMessage]
    public let startedAt: Date

    public init(
        id: UUID = UUID(),
        profileId: UUID,
        language: Language,
        kind: Kind,
        messages: [ChatMessage] = [],
        startedAt: Date = Date()
    ) {
        self.id = id
        self.profileId = profileId
        self.language = language
        self.kind = kind
        self.messages = messages
        self.startedAt = startedAt
    }

    public enum Kind: String, Codable, Sendable, CaseIterable {
        case chat
        case roast
        case dailyChallenge
        case partyQuestions
        case gameCoach
        case translate

        public var displayName: String {
            switch self {
            case .chat:           return "Chat"
            case .roast:          return "Roast"
            case .dailyChallenge: return "Daily Challenge"
            case .partyQuestions: return "Party Questions"
            case .gameCoach:      return "Game Coach"
            case .translate:      return "Translate"
            }
        }

        /// Whether this mode is shown in Kid-safe profile.
        public var availableInKidSafe: Bool {
            switch self {
            case .roast: return false
            default: return true
            }
        }
    }
}
